#151 Single Config

hunterwilliams opened the pull request:
For issue #151

grtjn commented on app/templates/README.mdown:
```diff
@@ -42,9 +42,9 @@ On Windows:
     ml.bat local deploy content
 
-Edit `gulp.config.js` to set your desired ports
+Edit `./deploy/gulp-local.json` to set your desired ports
```
you can put the gulp-local.json in the root, instead of under deploy/

grtjn commented on app/templates/gulpfile.js:
```diff
@@ -473,17 +482,27 @@ function serve(isDev, specRunner) {
+}
+
+function getNodeOptions(env) {
+  var envJson = require('./deploy/gulp-' + env + '.json');
```
Won't the require break if gulp-local.json doesn't exist? If yes, perhaps try to ignore the error, and maybe print a message suggesting you can provide local overrides with that file.

grtjn commented on slushfile.js:
```diff
@@ -260,34 +273,14 @@ gulp.task('generateSecret', ['init'], function(done) {
 
   try {
+    var configJSON = {};
+    configJSON['ml-host'] = 'localhost';
```
Can you make the slush generator ask for the host as well?

hunterwilliams commented:
@grtjn I've added the change so it doesn't need the xccport for ml8.

also I renamed ml-port to ml-http-port because it's a good clarification I think for inexperienced users

grtjn commented on app/templates/gulpfile.js:
```diff
@@ -439,12 +448,12 @@ function orderSrc (src, order) {
  * --debug-brk or --debug
  * --nosync
- * @param  {Boolean} isDev - dev or build mode
+ * @param  {String} env - 'dev','test', or 'prod'
```
Small typo? env = local | dev | prod..

grtjn commented on slushfile.js:
```diff
@@ -260,34 +289,19 @@ gulp.task('generateSecret', ['init'], function(done) {
+    fs.writeFileSync('gulp-local.json', configString, { encoding: 'utf8' });
   } catch (e) {
-    console.log('failed to update node-express-service: ' + e.message);
+    console.log('failed to write gulp.local.json: ' + e.message);
```
Tiny typo, gulp-local.json..

grtjn commented on app/templates/gulpfile.js:
```diff
@@ -473,17 +482,35 @@ function serve(isDev, specRunner) {
+    }
+    catch (e) {
+      envJson = {};
+      console.log('Info: ' + envFile + ' - may not exist. Did you mean to create one?');
```
A different project was using a similar pattern for a `local.gradle` file. If missing it printed:

"Couldn't find local.gradle; you can create this gitignore'd file to override properties if necessary"

Maybe print a similar message in our case?

grtjn replied:
A thought just struck me. People will ask themselves, how should gulp-local.json look like? How about adding a gulp init-local task that helps create that file, and refer to that in the info message?

grtjn commented on slushfile.js:
```diff
@@ -260,34 +289,19 @@ gulp.task('generateSecret', ['init'], function(done) {
 
   try {
+    var configJSON = {};
+    configJSON['ml-host'] = settings.marklogicHost;
```
tiny favor: can you add ml-version in here too?

grtjn commented:
Good work, really looking forward to this.. :)

hunterwilliams commented:
This does not cover proxy.js and routes.js properly for port changing looking into that now

hunterwilliams commented:
Fixed that. The change to of env.ML_PORT to env.ML_HTTP_PORT broke it as I wasn't populating it in getNodeOptions(). I changed it back since in getNodeOptions() it looks for env.ML_PORT and I'm not sure where that value is supposed to come from

Thoughts?

grtjn commented:
env.ML_PORT is used between gulpfile.js and node-app.js mostly. Looks good enough as is.

There seems to be a merge conflict, can you rebase? I'll have a spin with these changes after rebase..

grtjn commented:
Yay, didn't need to change generated local.properties and gulp-local.json to run ml local bootstrap and gulp serve! Really happy with this..

Just need to remind myself to not run gulp serve-dev, but `gulp serve-local` from now on!
